python/semantic_kernel/agents/bedrock/bedrock_agent.py

```diff
@@ -0,0 +1,406 @@
+
+        bedrock_agent.id = bedrock_agent.agent_model.agent_id
+        bedrock_agent.name = bedrock_agent.agent_model.agent_name
+
+        return bedrock_agent
+
+    async def create_agent(
+        self,
+        foundation_model: str,
+        role_arn: str,
+        enable_code_interpreter: bool | None = None,
+        enable_user_input: bool | None = None,
+        enable_kernel_function: bool | None = None,
+        **kwargs,
+    ) -> None:
+        """Create an agent asynchronously."""
+        await self._create_agent(
+            self.name,
+            foundation_model,
+            role_arn,
+            self.instructions,
+            **kwargs,
+        )
+
+        if enable_code_interpreter:
+            await self._create_code_interpreter_action_group()
+        if enable_user_input:
+            await self._create_user_input_action_group()
+        if enable_kernel_function:
+            await self._create_kernel_function_action_group()
+
+        await self._prepare_agent()
+        self.id = self.agent_model.agent_id
+
+    async def update_agent(
+        self,
+        agent_id,
+        agent_name,
+        role_arn,
+        foundation_model,
+        **kwargs,
+    ) -> BedrockAgentModel:
+        """Update an agent asynchronously."""
+        return await self._update_agent(
+            agent_id,
+            agent_name,
+            role_arn,
+            foundation_model,
+            **kwargs,
+        )
+
+    async def delete_agent(self, **kwargs) -> None:
+        """Delete an agent asynchronously."""
+        await self._delete_agent(**kwargs)
+
+    async def create_code_interpreter_action_group(self, **kwargs) -> BedrockActionGroupModel:
+        """Enable code interpretation."""
+        return await self._create_code_interpreter_action_group(**kwargs)
+
+    async def create_user_input_action_group(self, **kwargs) -> BedrockActionGroupModel:
+        """Enable user input."""
+        return await self._create_user_input_action_group(**kwargs)
+
+    async def create_kernel_function_action_group(self, **kwargs) -> BedrockActionGroupModel:
+        """Enable kernel function."""
+        return await self._create_kernel_function_action_group(self.kernel, **kwargs)
+
+    async def associate_agent_knowledge_base(self, knowledge_base_id, **kwargs) -> dict[str, Any]:
+        """Associate an agent with a knowledge base."""
+        return await self._associate_agent_knowledge_base(knowledge_base_id, **kwargs)
+
+    async def disassociate_agent_knowledge_base(self, knowledge_base_id, **kwargs) -> None:
+        """Disassociate an agent with a knowledge base."""
+        return await self._disassociate_agent_knowledge_base(knowledge_base_id, **kwargs)
+
+    async def list_associated_agent_knowledge_bases(self, **kwargs) -> dict[str, Any]:
+        """List associated agent knowledge bases."""
+        return await self._list_associated_agent_knowledge_bases(**kwargs)
+
+    async def invoke(
```

Comment:
Our agents should support both kernel arguments and kernel invoke-time overrides. Reference for handling those items: https://github.com/microsoft/semantic-kernel/blob/d0157424b4600a65ad58d699250b80c7162c17e4/python/semantic_kernel/agents/chat_completion/chat_completion_agent.py#L121C9-L127C52